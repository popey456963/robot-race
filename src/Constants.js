export const MOVE_ONE = 'MOVE_ONE'
export const MOVE_TWO = 'MOVE_TWO'
export const MOVE_THREE = 'MOVE_THREE'
export const BACK_UP = 'BACK_UP'
export const ROTATE_RIGHT = 'ROTATE_RIGHT'
export const ROTATE_LEFT = 'ROTATE_LEFT'
export const U_TURN = 'U_TURN'

export const NORTH = 'NORTH'
export const EAST = 'EAST'
export const SOUTH = 'SOUTH'
export const WEST = 'WEST'

export const PLAIN = 'PLAIN'
export const FLAG = 'FLAG'
export const HOLE = 'HOLE'
export const EXPRESS_CONVEYOR = 'EXPRESS_CONVEYOR'
export const CONVEYOR = 'CONVEYOR'

export const CONVEYOR_INDEX = 1
export const FLAG_INDEX = 2
export const EXPRESS_CONVEYOR_INDEX = 3
export const PLAIN_INDEX = 0

// export const TILE_SCALE = 16
// export const TILE_OFFSET = 6.4
// export const TILE_SIZE = 186

