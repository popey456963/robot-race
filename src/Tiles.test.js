import { createDirectionObject, genericTileObject, createFlagTile, createGrillTile, createHoleTile, createGearTile, createConveyorTile, createFastConveyorTile, createOutOfBoundsTile, createPlainTile, isTile } from './Tiles'
import { NORTH, EAST, SOUTH, WEST, CLOCKWISE, ANTICLOCKWISE, PLAIN, CONVEYOR, FLAG } from './Constants'

let directions = [
    undefined,
    [],
    [NORTH],
    [NORTH, EAST],
    [NORTH, SOUTH, WEST],
    [NORTH, SOUTH, EAST, WEST],
    [WEST, NORTH, SOUTH, EAST],
]

test('create direction object', () => {
    for (let direction of directions) {
        expect(createDirectionObject(direction)).toMatchSnapshot()
    }
})

test('generic tile object', () => {
    for (let direction of directions) {
        expect(genericTileObject({
            walls: direction,
            lasers: direction,
            pushers: direction
        })).toMatchSnapshot()
    }
})

test('flag tile object', () => {
    expect(createFlagTile({}, { flagNumber: 1 })).toMatchSnapshot()
    expect(createFlagTile({}, { flagNumber: 3 })).toMatchSnapshot()
    expect(createFlagTile({}, { flagNumber: 6 })).toMatchSnapshot()
})

test('grill tile object', () => {
    expect(createGrillTile({}, { level: 1 })).toMatchSnapshot()
    expect(createGrillTile({}, { level: 2 })).toMatchSnapshot()
})

test('hole tile object', () => {
    expect(createHoleTile({})).toMatchSnapshot()
})

test('gear tile object', () => {
    expect(createGearTile({})).toMatchSnapshot()
    expect(createGearTile({}, { rotationDirection: CLOCKWISE })).toMatchSnapshot()
    expect(createGearTile({}, { rotationDirection: ANTICLOCKWISE })).toMatchSnapshot()
})

test('conveyor tile object', () => {
    expect(createConveyorTile({})).toMatchSnapshot()
    expect(createConveyorTile({}, { exitDirection: NORTH, inputDirections: [NORTH, SOUTH] })).toMatchSnapshot()
    expect(createConveyorTile({}, { exitDirection: SOUTH, inputDirections: [] })).toMatchSnapshot()
})

test('fast conveyor tile object', () => {
    expect(createFastConveyorTile({})).toMatchSnapshot()
    expect(createFastConveyorTile({}, { exitDirection: NORTH, inputDirections: [NORTH, SOUTH] })).toMatchSnapshot()
    expect(createFastConveyorTile({}, { exitDirection: SOUTH, inputDirections: [] })).toMatchSnapshot()
})

test('out of bounds tile', () => {
    expect(createOutOfBoundsTile({ x: -1, y: -1 })).toMatchSnapshot()
})

test('is tile', () => {
    const plainTile = createPlainTile()

    expect(isTile(plainTile, [PLAIN])).toBe(true)
    expect(isTile(plainTile, [CONVEYOR])).toBe(false)

    const conveyorTile = createConveyorTile({}, { exitDirection: NORTH, inputDirections: [] })

    expect(isTile(conveyorTile, [CONVEYOR])).toBe(true)
    expect(isTile(conveyorTile, [CONVEYOR, PLAIN])).toBe(true)
    expect(isTile(conveyorTile, [PLAIN, FLAG])).toBe(false)
    expect(isTile(conveyorTile, [])).toBe(false)
})