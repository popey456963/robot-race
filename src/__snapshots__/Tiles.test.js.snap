// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conveyor tile object 1`] = `
Object {
  "lasers": Object {},
  "meta": Object {
    "exitDirection": undefined,
    "inputDirections": undefined,
  },
  "pushers": Object {},
  "type": "CONVEYOR",
  "walls": Object {},
}
`;

exports[`conveyor tile object 2`] = `
Object {
  "lasers": Object {},
  "meta": Object {
    "exitDirection": "NORTH",
    "inputDirections": Array [
      "NORTH",
      "SOUTH",
    ],
  },
  "pushers": Object {},
  "type": "CONVEYOR",
  "walls": Object {},
}
`;

exports[`conveyor tile object 3`] = `
Object {
  "lasers": Object {},
  "meta": Object {
    "exitDirection": "SOUTH",
    "inputDirections": Array [],
  },
  "pushers": Object {},
  "type": "CONVEYOR",
  "walls": Object {},
}
`;

exports[`create direction object 1`] = `Object {}`;

exports[`create direction object 2`] = `Object {}`;

exports[`create direction object 3`] = `
Object {
  "NORTH": true,
}
`;

exports[`create direction object 4`] = `
Object {
  "EAST": true,
  "NORTH": true,
}
`;

exports[`create direction object 5`] = `
Object {
  "NORTH": true,
  "SOUTH": true,
  "WEST": true,
}
`;

exports[`create direction object 6`] = `
Object {
  "EAST": true,
  "NORTH": true,
  "SOUTH": true,
  "WEST": true,
}
`;

exports[`create direction object 7`] = `
Object {
  "EAST": true,
  "NORTH": true,
  "SOUTH": true,
  "WEST": true,
}
`;

exports[`fast conveyor tile object 1`] = `
Object {
  "lasers": Object {},
  "meta": Object {
    "exitDirection": undefined,
    "inputDirections": undefined,
  },
  "pushers": Object {},
  "type": "FAST_CONVEYOR",
  "walls": Object {},
}
`;

exports[`fast conveyor tile object 2`] = `
Object {
  "lasers": Object {},
  "meta": Object {
    "exitDirection": "NORTH",
    "inputDirections": Array [
      "NORTH",
      "SOUTH",
    ],
  },
  "pushers": Object {},
  "type": "FAST_CONVEYOR",
  "walls": Object {},
}
`;

exports[`fast conveyor tile object 3`] = `
Object {
  "lasers": Object {},
  "meta": Object {
    "exitDirection": "SOUTH",
    "inputDirections": Array [],
  },
  "pushers": Object {},
  "type": "FAST_CONVEYOR",
  "walls": Object {},
}
`;

exports[`flag tile object 1`] = `
Object {
  "lasers": Object {},
  "meta": Object {
    "flagNumber": 1,
  },
  "pushers": Object {},
  "type": "FLAG",
  "walls": Object {},
}
`;

exports[`flag tile object 2`] = `
Object {
  "lasers": Object {},
  "meta": Object {
    "flagNumber": 3,
  },
  "pushers": Object {},
  "type": "FLAG",
  "walls": Object {},
}
`;

exports[`flag tile object 3`] = `
Object {
  "lasers": Object {},
  "meta": Object {
    "flagNumber": 6,
  },
  "pushers": Object {},
  "type": "FLAG",
  "walls": Object {},
}
`;

exports[`gear tile object 1`] = `
Object {
  "lasers": Object {},
  "meta": Object {
    "rotationDirection": undefined,
  },
  "pushers": Object {},
  "type": "GEAR",
  "walls": Object {},
}
`;

exports[`gear tile object 2`] = `
Object {
  "lasers": Object {},
  "meta": Object {
    "rotationDirection": "CLOCKWISE",
  },
  "pushers": Object {},
  "type": "GEAR",
  "walls": Object {},
}
`;

exports[`gear tile object 3`] = `
Object {
  "lasers": Object {},
  "meta": Object {
    "rotationDirection": "ANTICLOCKWISE",
  },
  "pushers": Object {},
  "type": "GEAR",
  "walls": Object {},
}
`;

exports[`generic tile object 1`] = `
Object {
  "lasers": Object {},
  "pushers": Object {},
  "walls": Object {},
}
`;

exports[`generic tile object 2`] = `
Object {
  "lasers": Object {},
  "pushers": Object {},
  "walls": Object {},
}
`;

exports[`generic tile object 3`] = `
Object {
  "lasers": Object {
    "NORTH": true,
  },
  "pushers": Object {
    "NORTH": true,
  },
  "walls": Object {
    "NORTH": true,
  },
}
`;

exports[`generic tile object 4`] = `
Object {
  "lasers": Object {
    "EAST": true,
    "NORTH": true,
  },
  "pushers": Object {
    "EAST": true,
    "NORTH": true,
  },
  "walls": Object {
    "EAST": true,
    "NORTH": true,
  },
}
`;

exports[`generic tile object 5`] = `
Object {
  "lasers": Object {
    "NORTH": true,
    "SOUTH": true,
    "WEST": true,
  },
  "pushers": Object {
    "NORTH": true,
    "SOUTH": true,
    "WEST": true,
  },
  "walls": Object {
    "NORTH": true,
    "SOUTH": true,
    "WEST": true,
  },
}
`;

exports[`generic tile object 6`] = `
Object {
  "lasers": Object {
    "EAST": true,
    "NORTH": true,
    "SOUTH": true,
    "WEST": true,
  },
  "pushers": Object {
    "EAST": true,
    "NORTH": true,
    "SOUTH": true,
    "WEST": true,
  },
  "walls": Object {
    "EAST": true,
    "NORTH": true,
    "SOUTH": true,
    "WEST": true,
  },
}
`;

exports[`generic tile object 7`] = `
Object {
  "lasers": Object {
    "EAST": true,
    "NORTH": true,
    "SOUTH": true,
    "WEST": true,
  },
  "pushers": Object {
    "EAST": true,
    "NORTH": true,
    "SOUTH": true,
    "WEST": true,
  },
  "walls": Object {
    "EAST": true,
    "NORTH": true,
    "SOUTH": true,
    "WEST": true,
  },
}
`;

exports[`grill tile object 1`] = `
Object {
  "lasers": Object {},
  "meta": Object {
    "level": 1,
  },
  "pushers": Object {},
  "type": "GRILL",
  "walls": Object {},
}
`;

exports[`grill tile object 2`] = `
Object {
  "lasers": Object {},
  "meta": Object {
    "level": 2,
  },
  "pushers": Object {},
  "type": "GRILL",
  "walls": Object {},
}
`;

exports[`hole tile object 1`] = `
Object {
  "lasers": Object {},
  "pushers": Object {},
  "type": "HOLE",
  "walls": Object {},
}
`;

exports[`out of bounds tile 1`] = `
Object {
  "position": Object {
    "x": -1,
    "y": -1,
  },
  "type": "OUT_OF_BOUNDS",
}
`;
